src/game/N3UIDebug.cpp

```diff
@@ -0,0 +1,209 @@
+
+    // Foreground window style: we don't need it visible, so disable some props.
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 0.0f);
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize, 0.0f);
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(0.0f, 0.0f));
+
+    // Begin
+    ImGui::Begin("Dockspace", NULL, s_DockWindowFlags);
+    ImGui::PopStyleVar();
+    ImGui::PopStyleVar(2);
+
+    ImGui::DockSpace(ImGui::GetID("MyDockspace"), ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_PassthruCentralNode);
+
+    ImGui::End();
+}
+
+void CN3UIDebug::RenderGameMetrics() {
+    ImGui::Begin("Game Metrics##metrics");
+
+    RenderFPSGraph();
+
+    ImGui::Separator();
+    ImGui::Text("Camera:");
+    CN3Camera * pCamera = CGameProcedure::s_pEng->CameraGetActive();
+    ImGui::BulletText("%-8.2f Field of View", D3DXToDegree(pCamera->m_Data.fFOV), 5, 5, " ");
+    ImGui::BulletText("%-8.2f Near Plane", pCamera->m_Data.fNP, 5, 5, " ");
+    ImGui::BulletText("%-8.2f Far Plane", pCamera->m_Data.fFP, 5, 5, " ");
+    ImGui::Checkbox("Fog Enabled", (bool *)&pCamera->m_bFogUse);
+    ImGui::DragInt("View Distance", &CN3Base::s_Options.iViewDist, 0.5f, 0, 10000);
+    if (ImGui::DragFloat("Field of View", &pCamera->m_Data.fFOV, 0.001f, 0.001f, 15.0f)) {
+        CN3Base::s_CameraData.fFOV = pCamera->m_Data.fFOV;
+    }
+    if (ImGui::DragFloat("Near Plane", &pCamera->m_Data.fNP, 0.001f, 0.01f, 5.0f)) {
+        CN3Base::s_CameraData.fNP = pCamera->m_Data.fNP;
+    }
+
+    ImGui::Separator();
+    ImGui::Text("Character:");
+    ImGui::BulletText("%-8d Character", CN3Base::s_RenderInfo.nChr);
+    ImGui::BulletText("%-8d Part", CN3Base::s_RenderInfo.nChr_Part);
+    ImGui::BulletText("%-8d Polygon", CN3Base::s_RenderInfo.nChr_Polygon);
+    ImGui::BulletText("%-8d Plug", CN3Base::s_RenderInfo.nChr_Plug);
+    ImGui::BulletText("%-8d Plug Polygon", CN3Base::s_RenderInfo.nChr_Plug_Polygon);
+
+    if (CGameBase::ACT_WORLD) {
+        ImGui::Separator();
+        ImGui::Text("Terrain:");
+        ImGui::BulletText("%-8d Polygon", CN3Base::s_RenderInfo.nTerrain_Polygon);
+        ImGui::BulletText("%-8d Tile Polygon", CN3Base::s_RenderInfo.nTerrain_Tile_Polygon);
+        ImGui::BulletText("%-8d Shape Count", CN3Base::s_RenderInfo.nShape);
+        ImGui::BulletText("%-8d Shape Part", CN3Base::s_RenderInfo.nShape_Part);
+        ImGui::BulletText("%-8d Shape Polygon", CN3Base::s_RenderInfo.nShape_Polygon);
+    }
+
+    ImGui::End();
+}
+
+void CN3UIDebug::RenderFPSGraph() {
+    static std::vector<float> s_Frames = {60.0f};
+
+    float fTime = CN3Base::TimeGet();
+
+    static float s_fTimePrev = fTime;
+    static float s_fTimeInterval = 0.0f;
+    static float s_fCurFrmPerSec = CN3Base::s_fFrmPerSec;
+    static float s_fFrmAvg = s_Frames[0];
+    static float s_fFrmMax = s_Frames[0];
+
+    static bool s_bPauseFrms = false;
+    if (!s_bPauseFrms) {
+        s_fTimeInterval += fTime - s_fTimePrev;
+        if (s_fTimeInterval >= 0.25f) {
```

Comment:
I believe editing the delay 0.25f would be a nice addition to this window. Similarly for frames size; instead of using `50` as fixed value, you can use `ImGui::InputInt`.

Reply:
Awesome idea! Fixed here: https://github.com/ko4life-net/ko/pull/168/commits/91b665f3fda40b0dc87fbda0d23057d669a417c0